### PR TITLE
[tools] Add meshcat display server for drake_visualizer's protocol

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
The server program is named "meldis".

Update the acrobot example's README to use it.

---

We need to port our demos away from `drake-visualizer` (#16216).  For many of those demos, viewing _only_ the geometry is sufficient (i.e., no need to show contact points, images, time, etc.).  This tool will allow us to immediately switch all of them to meshcat with changes to their README files, without any other code changes.

I've found it awkward to try to port their code directly to meshcat -- we need to change all of them to launch meshcat, then pause allow the user to click on the http link, and then begin the simulation.  Those tend to be more invasive changs that are a bit difficult to do quickly.  Meshcat works well in that mode for Jupyter notebooks, but for standalone C++ simulations the ergonomics are still underway.  This tool provides a nice stepping stone to help ease the transition.

Users who have their own C++ simulations will especially benefit.  They can switch to Meshcat as a display, without rewriting any of their code.

I have a follow-up PR on the way to add point-pair contact arrows as another applet.  If we end up liking this tool, we can investigate hydroelastic visualizations as well.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263)

### CODE REVIEW

**Review by sherm1 — COMMENTED** [review]
Cool!

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263#-:-MrOehzK-rS0FRqMQzuZ:b-89q77x)* status: 2 unresolved discussions, needs platform reviewer assigned, needs at least two assigned reviewers, labeled "do not merge", missing label for release notes (waiting on @jwnimmer-tri and @sherm1)

___
*[bindings/pydrake/manipulation/BUILD.bazel, line 89 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MrOec_ZCXmH7J4z5GEn:-MrOec_ZCXmH7J4z5GEo:b-rydpgk) ([raw file](https://github.com/robotlocomotion/drake/blob/8752e8127690a1aa4961b7a1446e7f2b9af69293/bindings/pydrake/manipulation/BUILD.bazel#L89)):*
> ```Starlark
> 
> drake_py_binary(
>     name = "fred",
> ```

BTW @mitiguy will love your choice of name
___
*[bindings/pydrake/manipulation/fred.py, line 3 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MrOeTQaEOLi81rpAECI:-MrOeTQaEOLi81rpAECJ:b-cdemeq) ([raw file](https://github.com/robotlocomotion/drake/blob/8752e8127690a1aa4961b7a1446e7f2b9af69293/bindings/pydrake/manipulation/fred.py#L3)):*
> ```Python
> """
> Offers a Meshcat vizualization server that listens for and draws Drake's
> legacy LCM vizualization messages.
> ```

BTW viz... -> visualization

**Comment by jwnimmer-tri** [comment]
+@rpoyner-tri for feature review, please.

This is introducing a new display program (see PR overview for details), so I'll also \CC @EricCousineau-TRI @ggould-tri @joemasterjohn @RussTedrake @sammy-tri @xuchenhan-tri for your thoughts on this approach / documentation / ergonomics in case you are interested.

**Review by xuchenhan-tri — COMMENTED** [review]
Thanks for the heads-up. I played with it and like it! Happy to help with hydro visualization if there's need for that.

> Reviewed 2 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263#-:-MuCKcjeCfSttT9RsvRP:b-21vg7v)* status: 1 unresolved discussion, LGTM missing from assignee rpoyner-tri(platform), needs at least two assigned reviewers, missing label for release notes (waiting on @jwnimmer-tri and @rpoyner-tri)

___
*[bindings/pydrake/visualization/meldis.py, line 16 at r3](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuCJJmzBFjk1Zdgh6w0:-MuCJJmzBFjk1Zdgh6w1:b-5to6ej) ([raw file](https://github.com/robotlocomotion/drake/blob/5573f32830aebd504cb44c9b6dd7e92a18075276/bindings/pydrake/visualization/meldis.py#L16)):*
> ```Python
> From a Drake binary release, run this as:
> 
>   python3 -m pydrake.visualization.meldis
> ```

Do we have any python examples that require this server (instead of directly going to meshcat)?
___
*[bindings/pydrake/visualization/meldis.py, line 18 at r3](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuCChJXBz-6W2kBEXrB:-MuCChJY4K-E5_dWY6BD:b-1ko794) ([raw file](https://github.com/robotlocomotion/drake/blob/5573f32830aebd504cb44c9b6dd7e92a18075276/bindings/pydrake/visualization/meldis.py#L18)):*
> ```Python
>   python3 -m pydrake.visualization.meldis
> 
> In many cases, pasing ``-w`` (i.e., ``--open-window``) to the program will be
> ```

nit, s/pasing/passing.

**Review by rpoyner-tri — COMMENTED** [review]
:lgtm: so far

> Reviewed 1 of 2 files at r2, 6 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263#-:-MuCOEuh5Z5LfS2KTvVa:b-vlbdz2)* status: 1 unresolved discussion, needs at least two assigned reviewers, missing label for release notes (waiting on @jwnimmer-tri)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuCMgEx0XHxS9Qe_3DW:-MuCMgEx0XHxS9Qe_3DX:b-8nsfxj) (no related file):*
Already discussed f2f -- I will probably want to run meldis remotely and the browser (window) locally. This may require control of port numbers from the command line.

---
*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuCO4ctEc7KPHLMmK1d:-MuCO4ctEc7KPHLMmK1e:b-vj3gur) (no related file):*
Already discussed f2f -- some way for meldis to notice that its browser has gone and shut down?

---

**Review by rpoyner-tri — COMMENTED** [review]
> Reviewed 1 of 8 files at r3.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263)* status: 1 unresolved discussion, needs at least two assigned reviewers, missing label for release notes (waiting on @jwnimmer-tri)

**Review by rpoyner-tri — COMMENTED** [review]
> Reviewed 1 of 8 files at r3.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263)* status: 1 unresolved discussion, needs at least two assigned reviewers, missing label for release notes (waiting on @jwnimmer-tri)

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263)* status: needs at least two assigned reviewers, missing label for release notes (waiting on @rpoyner-tri and @xuchenhan-tri)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuCO4ctEc7KPHLMmK1d:-MuCOzjW54VJ8MYm1yi5:b-1o5meb) (no related file):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

Already discussed f2f -- some way for meldis to notice that its browser has gone and shut down?
</blockquote></details>

Done via TODO.

---
*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuCMgEx0XHxS9Qe_3DW:-MuCPo4I3vazVHSgH7FT:b-896fix) (no related file):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

Already discussed f2f -- I will probably want to run meldis remotely and the browser (window) locally. This may require control of port numbers from the command line.
</blockquote></details>

Done.

---
___
*[bindings/pydrake/visualization/meldis.py, line 16 at r3](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuCJJmzBFjk1Zdgh6w0:-MuCQ2quC82wW4Z3EblM:b2e5147) ([raw file](https://github.com/robotlocomotion/drake/blob/5573f32830aebd504cb44c9b6dd7e92a18075276/bindings/pydrake/visualization/meldis.py#L16)):*
<details><summary><i>Previously, xuchenhan-tri wrote…</i></summary><blockquote>

Do we have any python examples that require this server (instead of directly going to meshcat)?
</blockquote></details>

Looks like pydrake.examples.multibody.cart_pole_passive_simulation is one example.  If you grep pydrake for `DrakeVisualizer` there's one or two more.

**Comment by jwnimmer-tri** [comment]
+@ggould-tri for platform review per schedule, please.

**Review by rpoyner-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263)* status: LGTM missing from assignee ggould-tri(platform) (waiting on @ggould-tri)

**Review by ggould-tri — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263)* status: LGTM missing from assignee ggould-tri(platform) (waiting on @jwnimmer-tri)

**Review by ggould-tri — COMMENTED** [review]
:lgtm:

> Reviewed 5 of 8 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263#-:-MuD1r376ioXJiDz3bvX:bnfp4nl)* status: 1 unresolved discussion (waiting on @jwnimmer-tri)

___
*[bindings/pydrake/visualization/meldis.py, line 71 at r4](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuD1eDv6V35wKsw_zkC:-MuD1eDv6V35wKsw_zkD:b-uurpw6) ([raw file](https://github.com/robotlocomotion/drake/blob/f656968aed91a06bcd8eb6233dfbe2bd7778c8cd/bindings/pydrake/visualization/meldis.py#L71)):*
> ```Python
>         self._meshcat.Delete(path=self._path)
> 
>         # Make ourselves invisible so that the viewer_load geometry does not
> ```

nit:  "make ourselves invisible" is a bit hard to parse:  Who is ourselves?  The viewer applet is clearly visible, after all.

"Make the loaded robot invisible" perhaps?

**Review by ggould-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263)* status: 2 unresolved discussions (waiting on @jwnimmer-tri)

___
*[-- commits, line 5 at r4](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuD22f03anRMIJ79reF):*
minor:  What do you mean by "update" here?  Wouldn't "consult" be the right verb?

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263)* status: :shipit: complete! all discussions resolved, LGTM from assignees rpoyner-tri(platform),ggould-tri(platform) (waiting on @ggould-tri and @rpoyner-tri)

___
*[-- commits, line 5 at r4](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuD22f03anRMIJ79reF):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

minor:  What do you mean by "update" here?  Wouldn't "consult" be the right verb?
</blockquote></details>

I'm using it the sense of "verb in the imperative", the same sense as "Add a display server", we have "Update the README".
___
*[bindings/pydrake/visualization/meldis.py, line 71 at r4](https://reviewable.io/reviews/robotlocomotion/drake/16263#-MuD1eDv6V35wKsw_zkC:-MuD5CmW2je0N8Aq7vbD:b-el6aql) ([raw file](https://github.com/robotlocomotion/drake/blob/f656968aed91a06bcd8eb6233dfbe2bd7778c8cd/bindings/pydrake/visualization/meldis.py#L71)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

nit:  "make ourselves invisible" is a bit hard to parse:  Who is ourselves?  The viewer applet is clearly visible, after all.

"Make the loaded robot invisible" perhaps?
</blockquote></details>

Done(?).

**Review by ggould-tri — COMMENTED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16263)* status: :shipit: complete! all discussions resolved, LGTM from assignees rpoyner-tri(platform),ggould-tri(platform) (waiting on @jwnimmer-tri)
